Map/LightOverlay.gd

```diff
@@ -0,0 +1,77 @@
+extends Node2D
+class_name LightOverlay
+
+var color_array: PackedColorArray
+
+var room_container: ColorRect
+var room_ui_array: Array[Array]
+
+var room_polygons_to_draw: Array[Rect2]
+
+func _init(_room_container: ColorRect, _room_ui_array: Array[Array]):
+	room_container = _room_container
+	room_ui_array = _room_ui_array
+
+func _draw():
+	var floor_rect_packed_array: PackedVector2Array
+	var room_circles: Array[PackedVector2Array]
+	var lit_rooms_to_draw: Array[RoomUI]
+	
+	# Account for minimum map placement T____T
+	for floor_index: int in range(room_ui_array.size()):
```

Comment:
why use a for with index instead of a for-loop on the values directly ?
I notice that you just use the index to get the actual floor, which could be summed up to:
`for floor: Array int room_ui_array`